Fix a missed dependency on ast.h which can result in race condition
at high parallel build

It fixed here

http://hg.python.org/cpython/rev/cfe0a293551f

[YOCTO #5884]

Upstream-Status: Backport
Signed-off-by: Khem Raj <raj.khem@gmail.com>

Index: Python-3.3.3/Makefile.pre.in
===================================================================
--- Python-3.3.3.orig/Makefile.pre.in	2014-02-26 09:33:13.868288863 -0800
+++ Python-3.3.3/Makefile.pre.in	2014-02-26 09:36:27.348292492 -0800
@@ -853,7 +853,8 @@
 		$(srcdir)/Include/warnings.h \
 		$(srcdir)/Include/weakrefobject.h \
 		pyconfig.h \
-		$(PARSER_HEADERS)
+		$(PARSER_HEADERS) \
+		$(AST_H)
 
 $(LIBRARY_OBJS) $(MODOBJS) Modules/python.o: $(PYTHON_HEADERS)
 
